### PR TITLE
Malcolm Cusack - The game so far

### PR DESCRIPTION
**Pull request by MalcolmCusack**
…Hey Sam, I created all the functions, i'm sure ill have to debug more. Can you look it over? I'm having trouble figuring out how to put everything together. It's like putting a puzzle together, but not knowing how to put together a puzzle. hahaha i'm going to come in early tomorrow as well. I compared my code to some other's code too.

### CODE REVIEW

**Comment by diiq** [comment]
Hi, Malcom. 

> It's like putting a puzzle together, but not knowing how to put together a puzzle

That's a very astute metaphor. You've tried to make a bunch of puzzle pieces without first planning out the whole picture. It's no surprise they don't seem to fit together!

We'll talk about this more in class, but remember how we did the squareRoot function? We started by writing out a simplified version of what we wanted, glossing over big details by calling on functions which didn't yet exist. So what's the high level description of our game? It could look something like:

```
game():
  check if user wants to play
  if so:
    set up the game
    play the game
    report the outcome
```

There are lots of ways you could break the game up, but that's one. Then you do the same thing to each of those sub-steps:

```
playGame:
  while the player hasn't won, lost, or quit:
    ask if the player wants to attack
    if so:
      deal damage
```

Eventually, you get to steps that are small enough to be easy to write -- and by that time, there's no puzzle about where it fits, because you're manufacturing a piece specifically to fit a whole in the picture.

In other words, design the big picture *first*, and then break it into little pieces. Don't try to write all the little pieces and then hope they're the right ones to assemble.

Again, you're not alone in this confusion, and we'll talk more about it in class.

File: labs/lab-2-javascript-game/script.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
+var playerHP = 40;
+var grantHP = 10;
+var playersName = "";
+var grantDeaths = 0;
+
+function startGame() {
+	 wouldYou = prompt("Would you like to play?");
```
Comment (diiq) [inline labs/lab-2-javascript-game/script.js:7]: Watch your indentation, here. That's a nine-space indent.

(I'd also recommend either 2 or 4 space indentation, rather than 8, which is what you're using in the rest of this file. 8 is not unheard-of, but it's a little eccentric. You can adjust this in your editor's settings.)

File: labs/lab-2-javascript-game/script.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+function startCombat() {
+	playerHP = 40;
+	grantDeaths = 0;
+	while (grantDeaths <= 3 && playerHP >= 0) {
+		 battle();
+	}
+	if (grantDeaths === 3) {
+	declareWinner();
+
+   }else if (playerHP <=0) {
```
Comment (diiq) [inline labs/lab-2-javascript-game/script.js:25]: Consistent spacing and indentation, please!

`} else if (playerHP <= 0) {`

File: labs/lab-2-javascript-game/script.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+ 	if (grantHP <= 0) {
+	grantDeaths = grantDeaths + 1;
+	console.log(playersName + ' has won a round');
+	}
+
+}
+
+function round() {
+	
+	attack = prompt('Do you want to attack? or quit?')
```
Comment (diiq) [inline labs/lab-2-javascript-game/script.js:45]: The instructions want you to ask me if I want to attack before each round of dealing damage, not before each round of killing.

File: labs/lab-2-javascript-game/script.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+	if (attack === 'attack') {
+
+		while (playerHP >= 0 && grantHP >= 0) {
```
Comment (diiq) [inline labs/lab-2-javascript-game/script.js:48]: What's this loop doing in here? It's not in your blueprint of round, in the comments below...

File: labs/lab-2-javascript-game/script.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+		startGame();
+	}
+}	
+
+function declareWinner() {
+	if (grantDeaths === 3) {
+		console.log(playersName + ' has won!!!');
+		playerHP = 40;
+		grantDeaths = 0;
+		startGame();
```
Comment (diiq) [inline labs/lab-2-javascript-game/script.js:68]: Don't immediately startGame. If I 'quit', I want to STOP playing.

**Review by diiq — COMMENTED** [review]
This is pretty rough, my pal. 

- Saying "quit" doesn't work.
- It appears to play until the user runs out of health, regardless of if they defeat grant 3 times.
- it asks if I want to attack and then attacks until someone dies, rather than asking about attacking each time.

Read the lab instructions carefully! Make a plan and then *follow the plan carefully*.

You should do some heavy studying about functions from your textbook before the assessment on saturday.
